app/controllers/coronavirus_form/address_lookup_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::AddressLookupController < ApplicationController
+  def show
+    @addresses = helpers.postcode_lookup(session[:postcode])
+    render controller_path, status: :ok
+  rescue AddressLookupError, AddressNotFoundError => e
+    session[:flash] = e.message
+
+    if e.class.name == "AddressLookupError" && e.status == 401
+      GovukError.notify("API Key is invalid!")
+      redirect_to support_address_path
+    else
+      redirect_to postcode_lookup_url
+    end
+  end
+
+  def submit
+    if session[:check_answers_seen]
+      redirect_to check_your_answers_url
+    else
+      address = helpers.uprn_lookup(params[:uprn])["results"][0]["LPI"]
```

Comment:
Nice spot on the dig, we've adjusted. On the 2nd point, we'd rather leave these as 2 separate calls. This makes it easier to test because it avoids having to chain methods together in the helper. We could maybe avoid this by having a third method along the lines of `lookup_address_by_uprn_and_convert` but I really don't like that.